simple-loadflow/src/main/java/com/powsybl/loadflow/simple/SimpleLoadFlow.java

```diff
@@ -45,47 +52,75 @@
 
     private final Network network;
 
+    private final MatrixFactory matrixFactory;
+
     public SimpleLoadFlow(Network network) {
+        this(network, new DenseMatrixFactory());
+    }
+
+    public SimpleLoadFlow(Network network, MatrixFactory matrixFactory) {
         this.network = Objects.requireNonNull(network);
+        this.matrixFactory = Objects.requireNonNull(matrixFactory);
+    }
+
+    private static void balance(NetworkContext networkContext) {
+        double activeGeneration = 0;
+        double activeLoad = 0;
+        for (Bus b : networkContext.getBuses()) {
+            for (Generator g : b.getGenerators()) {
+                activeGeneration += g.getTargetP();
+            }
+            for (Load l : b.getLoads()) {
+                activeLoad += l.getP0();
+            }
```

Comment:
Don't you miss dangling lines in the balance here ?

Reply:
Added in the balance but the problem is even more serious, dangling lines are not taken into account in the calculation. I just added an exception throw in case of dangling line in the network.